### PR TITLE
Support lineage option

### PR DESCRIPTION
**Pull request by Konboi**
Support `--lineage` option.

e.g)
```
$ launchable record session --build <BUILD NAME> --lineage feature-xxx
```

### CODE REVIEW

File: launchable/commands/helper.py [hunk 1]
```diff
@@ -16,6 +16,7 @@ def find_or_create_session(
     is_observation: bool = False,
     links: List[str] = [],
     is_no_build: bool = False,
+    lineage: Optional[str] = None,
 ) -> Optional[str]:
     """Determine the test session ID to be used.
```
Comment (shibuiwilliam) [inline launchable/commands/helper.py:22]: [nits]
I think it is a good chance that we can add missing args in this comment, from flavor to lineage.

File: launchable/commands/subset.py [hunk 2]
```diff
@@ -141,6 +141,14 @@
     help="If you want to only send test reports, please use this option",
     is_flag=True,
 )
+@click.option(
+    '--lineage',
+    'lineage',
+    help='Set lineage name',
```
Comment (shibuiwilliam) [inline launchable/commands/subset.py:147]: [must]
`lineage` is unclear term. It is better to explain what is lineage and how it changes subset command.

File: launchable/commands/record/tests.py [hunk 3]
```diff
@@ -143,6 +143,14 @@ def _validate_group(ctx, param, value):
     type=str,
     metavar='SESSION_NAME',
 )
+@click.option(
+    '--lineage',
+    'lineage',
+    help='Set lineage name',
```
Comment (shibuiwilliam) [inline launchable/commands/record/tests.py:149]: same here
https://github.com/launchableinc/cli/pull/586/files#r1266228933

File: launchable/commands/record/session.py [hunk 4]
```diff
@@ -82,6 +82,14 @@ def _validate_session_name(ctx, param, value):
     metavar='SESSION_NAME',
     callback=_validate_session_name,
 )
+@click.option(
+    '--lineage',
+    'lineage',
+    help='Set lineage name',
```
Comment (shibuiwilliam) [inline launchable/commands/record/session.py:88]: same here
https://github.com/launchableinc/cli/pull/586/files#r1266228933

File: launchable/commands/record/session.py [hunk 5]
```diff
@@ -132,6 +141,7 @@ def session(
         "flavors": flavor_dict,
         "isObservation": is_observation,
         "noBuild": is_no_build,
+        "lineage": lineage if lineage else "",
```
Comment (shibuiwilliam) [inline launchable/commands/record/session.py:144]: [question]
Is empty string handled the same as NULL in the backend side?
Is there no chance that a user can set "" lineage?
Is the empty "" lineage validated to fail somewhere?

**Review by shibuiwilliam — COMMENTED** [review]
commented.

File: launchable/commands/record/session.py (hunk 5 above)
Comment (Konboi) [reply launchable/commands/record/session.py:144]: We can set `None` at the cli (Python) side when it's empty. 
So, remove this `if` statement for easy to read.

File: launchable/commands/record/session.py [hunk 6]
```diff
@@ -82,6 +82,14 @@ def _validate_session_name(ctx, param, value):
     metavar='SESSION_NAME',
     callback=_validate_session_name,
 )
+@click.option(
+    '--lineage',
+    'lineage',
+    help='Set lineage name. A lineage is a set of test sessions grouped and this option value will be used for grouping.',
```
Comment (shibuiwilliam) [inline launchable/commands/record/session.py:88]: `this option value will be used for grouping.` 
Do you mean the lineage name becomes group name, or something else?

File: launchable/commands/record/tests.py [hunk 7]
```diff
@@ -143,6 +143,14 @@ def _validate_group(ctx, param, value):
     type=str,
     metavar='SESSION_NAME',
 )
+@click.option(
+    '--lineage',
+    'lineage',
+    help='Set lineage name. This option value will be passed the record session command if a session isn\'t created yet.',
```
Comment (shibuiwilliam) [inline launchable/commands/record/tests.py:149]: `This option value will be passed to the record session command`

File: launchable/commands/subset.py [hunk 8]
```diff
@@ -141,6 +141,14 @@
     help="If you want to only send test reports, please use this option",
     is_flag=True,
 )
+@click.option(
+    '--lineage',
+    'lineage',
+    help='Set lineage name. This option value will be passed the record session command if a session isn\'t created yet.',
```
Comment (shibuiwilliam) [inline launchable/commands/subset.py:147]: `This option value will be passed to the record session command`

File: tests/test_runners/test_dotnet.py [hunk 9]
```diff
@@ -94,6 +94,84 @@ def test_subset(self):
         output = "FullyQualifiedName!=rocket_car_dotnet.ExampleTest.TestAdd&FullyQualifiedName!=rocket_car_dotnet.ExampleTest.TestDiv\n"  # noqa: E501
         self.assertEqual(result.output, output)
 
+    @responses.activate
```
Comment (shibuiwilliam) [inline tests/test_runners/test_dotnet.py:97]: Is this change from the PR previously merged?
https://github.com/launchableinc/cli/pull/588

File: launchable/commands/record/session.py (hunk 6 above)
Comment (Konboi) [reply launchable/commands/record/session.py:88]: Ah... I should fix `this option value will be used for a lineage name`

File: tests/test_runners/test_dotnet.py (hunk 9 above)
Comment (Konboi) [reply tests/test_runners/test_dotnet.py]: I rebased with the main branch and fixed the diff.

**Review by shibuiwilliam — APPROVED** [review]

**Pull request merged**